src/main/java/org/opensearch/neuralsearch/processor/NeuralSparseTwoPhaseProcessor.java

```diff
@@ -0,0 +1,223 @@
+        // Make a nestedQueryBuilder which includes all the two-phase QueryBuilder.
+        QueryBuilder nestedTwoPhaseQueryBuilder = getNestedQueryBuilderFromNeuralSparseQueryBuilderMap(queryBuilderMap);
+        nestedTwoPhaseQueryBuilder.boost(getOriginQueryWeightAfterRescore(request.source()));
+        // Add it to the rescorer.
+        RescorerBuilder<QueryRescorerBuilder> twoPhaseRescorer = buildRescoreQueryBuilderForTwoPhase(nestedTwoPhaseQueryBuilder,request);
+        request.source().addRescorer(twoPhaseRescorer);
+        return request;
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Based on ratio, split a Map into two map by the value.
+     * @param queryTokens the queryTokens map, key is the token String, value is the score.
+     * @param ratio The ratio that control how tokens map be split.
+     * @return A map has two element, {[True, token map whose value above threshold],[False, token map whose value below threshold]}
+     */
+    public static Map<Boolean, SetOnce<Map<String, Float>>> getSplitSetOnceByScoreThreshold(
+        final Map<String, Float> queryTokens,
+        final float ratio
+    ) {
+        float max = 0f;
+        for (Float value : queryTokens.values())
+            max = Math.max(value, max);
+        float threshold = max * ratio;
+
+        Map<Boolean, Map<String, Float>> queryTokensByScore = queryTokens.entrySet()
+            .stream()
+            .collect(
+                Collectors.partitioningBy(entry -> entry.getValue() >= threshold, Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+            );
+        SetOnce<Map<String, Float>> highScoreTksSetOnce = new SetOnce<>(queryTokensByScore.get(Boolean.TRUE));
+        SetOnce<Map<String, Float>> lowScoreTksSetOnce = new SetOnce<>(queryTokensByScore.get(Boolean.FALSE));
+        if (highScoreTksSetOnce.get() == null) throw new IllegalArgumentException();
+        return Map.of(Boolean.TRUE, highScoreTksSetOnce, Boolean.FALSE, lowScoreTksSetOnce);
+    }
+
+    /**
+     * Factory to create NeuralSparseTwoPhaseProcessor, provide default parameter,
+     *
+     */
+    public static class Factory implements Processor.Factory<SearchRequestProcessor> {
+        @Override
+        public NeuralSparseTwoPhaseProcessor create(
+            Map<String, Processor.Factory<SearchRequestProcessor>> processorFactories,
+            String tag,
+            String description,
+            boolean ignoreFailure,
+            Map<String, Object> config,
+            PipelineContext pipelineContext
+        ) throws IllegalArgumentException {
+
+            boolean enabled = ConfigurationUtils.readBooleanProperty(TYPE, tag, config, ENABLE_KEY, true);
+            Map<String, Object> twoPhaseConfigMap = ConfigurationUtils.readOptionalMap(TYPE, tag, config, PARAMETER_KEY);
+
+            float ratio = 0.4f;
+            float window_expansion = 5.0f;
+            int max_window_size = 10000;
```

Comment:
please make constants for those 3 values you're assigning to variable initially

Reply:
Use constants for default value and camel case for variables.